Docs/developer_guide/python_faq.md

```diff
@@ -2,6 +2,85 @@
 
 Frequently asked questions about how to write Python scripts for Slicer.
 
+## What is the Slicer Python Environment?
+
+You can consider each Slicer installation as a virtual Python environment - the same way
+as you create virtual environments using python or conda.
+
+## What is the PythonSlicer executable?
+
+`PythonSlicer` is an executable provided in the `bin` directory of the Slicer installation.
+It’s the same version of Python available through the Slicer application, but it acts like
+a normal `python` (or `python3`) executable in terms of running scripts and can be used t
```

Comment:
PythonSlicer acts as a usual `python` executable in every sense, not just in term of running scripts. For example, it can be used as an interactive console, too. So, I'm not sure if we need to narrow its scope (we could just remove the "in term of..." part).

Reply:
Very good point, I revisited the wording:

```
`PythonSlicer` is an executable provided in the `bin` directory of the Slicer installation.
It is a Python interpreter that allows access to all of the packages installed in Slicer.
This makes it possible to use the Slicer Python environment from outside the application, such
as for batch processing, command-line operations, terminal-based interactive session and
integration with IDEs.

This means you can use `PythonSlicer` as a replacement for the regular Python interpreter
(`python` or `python3`) to take advantage of the installed packages in the Slicer environment. 

:::{warning}
Please note that the running Python context is **not** available when using `PythonSlicer`.
As a result, many objects, including the application instance, the MRML scene and the loaded
modules, are not available.
:::
```